### PR TITLE
RET-2758

### PR DESCRIPTION
**Pull request by ikirsanov**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-2758

### Change description ###
This story is to generate an acknowledgment email to the Claimant for a Type A/B/C application they submitted to contact the tribunal about something

NB As part of this story email templates required (as per attachments/figma) will need to either already exist/be created in GOV.UK Notify (https://www.notifications.service.gov.uk/):

Email should be generated by the system as follows:

For Type A/B applications

If Claimant has answered ‘Yes’ to Rule92 question receives acknowledgment email for their submitted application, with wording that Respondent will be notified
Email template to be used is dependent on whether it is a type A or type B application.

(Ref: figma link [Rule 92 - Copying correspondence to the other party – Figma](https://www.figma.com/file/lKOHBXVvhkxx80y81lSWmc/Rule-92---Copying-correspondence-to-the-other-party?node-id=0%3A1)  And Application content matrix (in attachments)

If Claimant has answered ‘No’ to Rule92 question receives acknowledgment email (and guidance on next steps due) for their submitted application
Email template to be used is the same for both Type A and B application

For Type C applications (Order a witness to attend)

Rule 92 does not apply for this type of application

Claimant receives acknowledgment email that application has been submitted

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by marcjudge — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/helper/NotificationsHelper.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        if (CollectionUtils.isEmpty(caseData.getRespondentCollection())) {
+            return StringUtils.EMPTY;
+        }
+        return caseData.getRespondentCollection().stream()
+            .map(o -> o.getValue().getRespondentName())
+            .collect(Collectors.joining(", "));
+    }
+
+    /**
+     * Gets the .
```
Comment (Itay-Kirsh) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/helper/NotificationsHelper.java:66]: Gets what

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 2]
```diff
@@ -97,4 +109,76 @@ public SendEmailResponse sendSubmitCaseConfirmationEmail(CaseDetails caseDetails
...
+
+        SendEmailResponse sendEmailResponse;
+        Map<String, String> parameters = new ConcurrentHashMap<>();
+        parameters.put("claimant", claimant);
+        parameters.put("respondentNames", respondentNames);
+        parameters.put("caseId", caseId);
+        parameters.put("caseNumber", caseNumber);
+
+        String emailTemplate = notificationsProperties.getClaimantTseEmailTypeCTemplateId();
+        if (!TYPE_C.equals(claimantApplication.getContactApplicationType())) {
```
Comment (Itay-Kirsh) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:138]: Very complex method, could we simplify it a little

**Review by Itay-Kirsh — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java (hunk 2 above)
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java]: Refactored and simplified

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/helper/NotificationsHelper.java (hunk 1 above)
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/helper/NotificationsHelper.java]: Fixed

**Review by Itay-Kirsh — APPROVED** [review]

**Review by brendanmchugh — APPROVED** [review]

**Pull request merged**